### PR TITLE
User allocation helper scripts

### PR DESCRIPTION
**Pull request by tschubotz**

### CODE REVIEW

File: user-allocations/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+csv
```
Comment (rmeissner) [inline user-allocations/.gitignore:1]: Is this supposed to ignore a folder called `csv` or what is the purpose of this one?

Comment (tschubotz) [reply user-allocations/.gitignore]: that's a forgotten leftover from a previous version. removed.

File: user-allocations/fetch_rewards.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+# This includes uncle rewards.
+
+from datetime import datetime
+import csv
+
+from etherscan.accounts import Account
+from etherscan.client import EmptyResponse
+
+
+ETHERSCAN_API_KEY = ''
```
[the remark this answers is not in the document]
Comment (tschubotz) [reply user-allocations/fetch_rewards.py]: done

File: user-allocations/fetch_rewards.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+MINER_SAFES = ['0xf20b338752976878754518183873602902360704','0xae5fb390e5c4fa1962e39e98dbfb0ed8055ed7a9']  # Use https://dune.com/queries/638270 to get this list.
+OUTPUT_FILENAME_SQL = './rewards.sql'
+OUTPUT_FILENAME_CSV = './rewards.csv'
+page_size = 10000  # Etherscan page size
+
+data = {}
+
+# Fetch data
+for address in MINER_SAFES:
+    etherscan_api = Account(address=address, api_key=ETHERSCAN_API_KEY)
```
[the remark this answers is not in the document]
Comment (tschubotz) [reply user-allocations/fetch_rewards.py]: yes, that package only supports mainnet. it's a very simple one.

File: user-allocations/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+- Fetches block and uncle rewards from Etherscan for a given list of addresses.
+- Outputs data as CSV and SQL to be put into a query on Dune.
+
+## fetch_sanctioned.py
+
+- For a list of addresses (Safes), checks whether they appear in the Chainalysis on-chain list of sanctioned addresses.
+
+## Scripts and queries 
+
+1. Install requirements via `pip install -r requirements.txt`
```
[the remark this answers is not in the document]
Comment (tschubotz) [reply user-allocations/README.md]: done

File: user-allocations/.env.sample [hunk 5]
```diff
@@ -0,0 +1,2 @@
+ETHERSCAN_API_KEY=''
```
Comment (rmeissner) [inline user-allocations/.env.sample:1]: Shouldn't we be able to reuse the existing env file? else the user has to maintain 2 different env files

File: user-allocations/.gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+.env
```
Comment (rmeissner) [inline user-allocations/.gitignore:1]: Shouldn't this already be covered by the gitignore in the root?

Comment (rmeissner) [inline user-allocations/.gitignore:1]: Shouldn't this already be covered by the gitignore in the root?

Comment (rmeissner) [inline user-allocations/.gitignore:1]: Shouldn't this already be covered by the gitignore in the root?

File: user-allocations/.env.sample (hunk 5 above)
Comment (tschubotz) [reply user-allocations/.env.sample:1]: yes

Comment (rmeissner) [inline user-allocations/.env.sample:1]: We should be able to remove this

File: user-allocations/README.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+## Scripts and queries 
+
+(Use a virtualenv if possible.)
+
+1. Create virtual env: `python -m venv venv`
+2. Activate virtual env `source venv/bin/activate`
+3. Install requirements via `pip install -r requirements.txt`
+4. Copy file with env variables: `cp ../.env.sample ../.env`
+5. Add you API keys to `.env`
```
Comment (rmeissner) [inline user-allocations/README.md:22]: Nitpic: should be `../.env` (if you want to be exact :P )

**Review by rmeissner — APPROVED** [review]

**Pull request merged**